lib/saml_idp/assertion_builder.rb

```diff
@@ -40,7 +42,7 @@ def fresh
           assertion.Issuer issuer_uri
           sign assertion
           assertion.Subject do |subject|
-            subject.NameID name_id, Format: name_id_format[:name]
+            subject.NameID name_id, Format: sp_name_id_format[:name]
```

Comment:
This really ought to be a `.fetch()` since bad things will happen if this key isn't there.